### ISSUE TITLE
Recent test infrastructure refactoring appears to have broken WinUI interaction tests 

### ISSUE THREAD

**Issue by Felix-Dev** [microsoft/microsoft-ui-xaml#3349]
The recent test infrastructure refactoring in PR https://github.com/microsoft/microsoft-ui-xaml/commit/c46d28b3e706a24a5b89ec110ac2e93a9a86aa58 seems to have broken the local test infrastructure for *interaction* tests on my local system (API tests still work fine). 

#### Using the Innerloop solution
Prior to this PR, I could run interaction tests just fine. I would build and run the solution which would install the *MUXControlsInnerLoopTestApp* on my system and then start interaction tests via Visual Studio's Test Explorer. The powershell process output to start an interaction test would look like this:
![image](https://user-images.githubusercontent.com/1398851/94586492-96dc7880-0281-11eb-96c5-1a188a4814ee.png)

No certificate installation required or anything - the test app would just be started and the interaction test executed.

Starting with PR https://github.com/microsoft/microsoft-ui-xaml/commit/c46d28b3e706a24a5b89ec110ac2e93a9a86aa58, however, I am now asked to install the certificate. If I don't do that, the test process fails with the message:
```
Failed to install AppX for MUXControlsTestApp!
```

This error occurs in `\microsoft-ui-xaml\test\testinfra\MUXTestInfra\Infra\Applications.cs` - InstallTestAppIfNeeded():
![image](https://user-images.githubusercontent.com/1398851/94589349-74e4f500-0285-11eb-9159-da4ab194323a.png)

Even if I do install the certificate - which I never had to install before - I get the following error:
![image](https://user-images.githubusercontent.com/1398851/94589593-bc6b8100-0285-11eb-93b6-68b5f500cbcf.png)

If I uninstall the existing MUXControlsInnerLoopTestApp and then run the tests, the powershell script apparently finishes successfully:
![image](https://user-images.githubusercontent.com/1398851/94590083-6f3bdf00-0286-11eb-9a3a-52311c5cec00.png)
However, a new test instance of the MUXControlsInnterLoopTestApp is never started - the interaction test never finishes:
![image](https://user-images.githubusercontent.com/1398851/94590189-9a263300-0286-11eb-9589-6759c683586f.png)

#### Using the full solution
Running interaction tests in the full solution (MUXControls.sln) also appears to have been broken with that PR. If I run interactions tests on the PR right before https://github.com/microsoft/microsoft-ui-xaml/commit/c46d28b3e706a24a5b89ec110ac2e93a9a86aa58, I can run the tests just fine. If, however, I run interaction tests with that PR, I now get the following error message in VS's test output window:
```
---------- Starting test discovery for requested test run ----------
Microsoft.VisualStudio.TestPlatform.ObjectModel.TestPlatformException: Testhost process exited with error: It was not possible to find any compatible framework version
The framework 'Microsoft.NETCore.App', version '2.1.0' was not found.
  - The following frameworks were found:
      3.1.7 at [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
      5.0.0-preview.5.20278.1 at [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
You can resolve the problem by installing the specified framework and/or SDK.
```
It doesn't actually matter if I have .NET Core 2.1.0 (in my case 2.1.801) installed or not, I always get the same error message. This never happened before.

To summarize:
Starting with PR https://github.com/microsoft/microsoft-ui-xaml/commit/c46d28b3e706a24a5b89ec110ac2e93a9a86aa58 I can no longer run interactions tests using both the full or innerloop solution. In other words, right now I am blocked on my current and future PRs which require interaction tests and are based on an up-to-date WinUI master branch.

@kmahone @azchohfi FYI.

**Comment by chingucoding** [comment]
While full solution interaction tests run fine, I am also unable to run interaction tests when using the MUXControlsInnerloop.sln file. As @Felix-Dev noted, this seemed to degrade with c46d28b .

**Comment by StephenLPeters** [comment]
@azchohfi and @kmahone looks like the test refactor work broke the inner loop project.

**Comment by Felix-Dev** [comment]
I am closing this as I can now run interaction tests in both the full solution as well as the innerloop solution again.
